Watchman.Discord.UnitTests/AntiSpam/OverallSpamDetectorTests.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using Devscord.DiscordFramework.Framework.Commands.AntiSpam;
+using Devscord.DiscordFramework.Framework.Commands.AntiSpam.Models;
+using Devscord.DiscordFramework.Services.Models;
+using Moq;
+using NUnit.Framework;
+using Watchman.Discord.Areas.Protection.Strategies;
+using Watchman.DomainModel.Messages.Queries;
+
+namespace Watchman.Discord.UnitTests.AntiSpam
+{
+    [TestFixture]
+    internal class OverallSpamDetectorTests
+    {
+        [Test]
+        [TestCase("abc", "xyz", "spam", "spam", false, SpamProbability.Low)]
+        [TestCase("xyz", "aaa", "abc", "https://discord.gg/example", true, SpamProbability.Low)]
+        [TestCase("abc", "xyz", "spam", "spam", true, SpamProbability.Low)]
+        [TestCase("xyz", "aaa", "abc", "https://discord.gg/example", false, SpamProbability.Medium)]
+        [TestCase("spam", "spam", "spam", "spam", true, SpamProbability.Medium)]
```

Comment:
czym w takim razie różni sie wykrywanie dla isUserSafe=true od isUserSafe=false? 

to nie jest tak że jeśli isUserSafe=true, to zawsze powinno ignorować?

Reply:
to jest dlatego, że przyjąłem, że safe user ma znacznie mniejsze szanse na bycie spamerem, ale nie odrzucam takiej możliwości
W tym systemie po prostu safe user musi się bardziej postarać aby dostać jakikolwiek punishment
Co o tym myślisz?

Reply:
chodziło mi bardziej o to, że co jeśli wykrywanie isUserSafe nie działa prawidłowo 
powinno być więcej przypadków sprawdzających wersje z isUserSafe

Reply:
to nie rozumiem
jest mniej więcej równo TestCase'ów dla true i false